core/types/bloom9.go

```diff
@@ -57,10 +57,8 @@ func (b *Bloom) SetBytes(d []byte) {
 }
 
 // Add adds d to the filter. Future calls of Test(d) will return true.
-func (b *Bloom) Add(d *big.Int) {
-	bin := new(big.Int).SetBytes(b[:])
-	bin.Or(bin, bloom9(d.Bytes()))
-	b.SetBytes(bin.Bytes())
+func (b *Bloom) Add(d []byte) {
+	b.SetBytes(or(b[:], bloom9(d)))
```

Comment:
Since `or` modifies the first param... and then you do SetBytes on `b` with `b[:]`... I find it very hard to figure out what the end result of this is... Is the SetBytes even needed? Don't you just copy the same buffer to itself?